update remote-data-services after fsspec [skip ci] (#5170)
dask/dask: docs/source/remote-data-services.rst

   import dask.bag as db
   b = db.read_text('hdfs://path/to/*.json').map(json.loads)

Dask uses `fsspec`_ for local, cluster and remote data IO. Other file interaction, such
as loading of configuration, is done using ordinary python method.

The following remote services are well supported and tested against the main
codebase:

- **Local or Network File System**: ``file://`` - the local file system, default in the absence of any protocol

- **Hadoop File System**: ``hdfs://`` - Hadoop Distributed File System, for resilient, replicated

>   files within a cluster. Uses pyarrow_ as the backend.

Reviewer: Just to make this a complete sentence

```suggestion
  files within a cluster. This uses pyarrow_ as the backend.
```

- **HTTP(s)**: ``http://`` or ``https://`` for reading data directly from HTTP web servers

- **Azure Datalake Storage**: ``adl://``, for use with the Microsoft
  Azure platform, using azure-data-lake-store-python_, is unavailable in the current release of
  ``fsspec``, but a new version using Microsoft's "protocol 2" should come soon.

``fsspec`` also provides other file sytstems that may be of interest to Dask users, such as

> ssh, ftp, webhdfs - see the documentation.

Reviewer: ```suggestion
ssh, ftp, and webhdfs.  See the documentation for more information.
```

Two methods exist for passing parameters to the backend file system driver:
extending the URL to include username, password, server, port, etc.; and
providing ``storage_options``, a dictionary of parameters to pass on. The
second form is more general, as any number of file system-specific options
can be passed.

> Examples:

Reviewer: Did you intend for `Examples` to be on a new line?

   df = dd.read_parquet('s3://bucket/path',
                        storage_options={'anon': True, 'use_ssl': False})

> Further details on how to provide configuration for the main back-ends

Reviewer: We say further details twice here

```suggestion 
Details on how to provide configuration for the main back-ends
```

> is listed next, but further details can be found in the docs pages of the

Reviewer: ```suggestion
is listed next, but further details can be found in the documentation of the
```

> relavant back-end.

Reviewer: ```suggestion
relevant back-end.
```

We should probably run spellcheck on the file as a whole

Each back-end has additional installation requirements and may not be available
at runtime. The dictionary ``fsspec.registry`` contains the

> currently imported file systems, and to see which backends ``fsspec`` knows how

Reviewer: ```suggestion
currently imported file systems.  To see which backends ``fsspec`` knows how
```

to import, you can do

.. code-block:: python

    from fsspec.registry import known_implementations
    known_implementations

Note that some backends appear twice, if they can be referenced with multiple

> protocol strings, e.g., "http" and "https".

Reviewer: Copy-editing people often yell at me for using "e.g.", which is apparently unclear for non-native english speakers

```suggestion
protocol strings, like "http" and "https".
```

Honestly, even as a native english speaker I'm not entirely sure what it means :)

The Hadoop File System (HDFS) is a widely deployed, distributed, data-local file
system written in Java. This file system backs many clusters running Hadoop and

> Spark. HDFS support can be provided by pyarrow_.

Reviewer: ```suggestion
Spark. HDFS support can be provided by PyArrow_.
```

Given that we're capitalizing Hadoop, Spark, and HDFS, we should probably capitalize PyArrow as well to show equal respect.
